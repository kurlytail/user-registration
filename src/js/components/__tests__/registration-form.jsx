import React from 'react';
import ReCAPTCHA from 'react-google-recaptcha';
import Renderer from 'react-test-renderer';
import RegistrationForm from '../registration-form';

jest.mock('react-google-recaptcha');

const FIXTURES = {
    actions: {
        getRegistration: jest.fn(),
        createRegistration: jest.fn()
    },

    nullState: null,

    fetchingState0: {},

    fetchingState1: {
        isFetching: true
    },

    fetchingState2: {
        item: {}
    },

    fetchingState3: {
        item: {},
        isFetching: true
    },

    fetchedState: {
        item: { captchaKey: 'test-key' },
        isFetching: false
    },

    emailEvent: {
        target: {
            value: 'test-email'
        }
    },

    submitEvent: {
        preventDefault: jest.fn()
    }
};

describe('# registration-form', () => {
    beforeEach(() => {
        Object.entries(FIXTURES.actions).forEach(([name, action]) => action.mockReset());
    });

    describe('## constructor', () => {
        it('### should create default object', () => {
            const component = Renderer.create(<RegistrationForm />);
            expect(component.toJSON()).toMatchSnapshot();
        });
    });

    describe('## render', () => {
        it('### render with different states', () => {
            let component = Renderer.create(<RegistrationForm state={FIXTURES.nullState} actions={FIXTURES.actions} />);
            expect(component.toJSON()).toMatchSnapshot();

            component = Renderer.create(
                <RegistrationForm state={FIXTURES.fetchingState0} actions={FIXTURES.actions} />
            );
            expect(component.toJSON()).toMatchSnapshot();

            component = Renderer.create(
                <RegistrationForm state={FIXTURES.fetchingState1} actions={FIXTURES.actions} />
            );
            expect(component.toJSON()).toMatchSnapshot();

            component = Renderer.create(
                <RegistrationForm state={FIXTURES.fetchingState2} actions={FIXTURES.actions} />
            );
            expect(component.toJSON()).toMatchSnapshot();

            component = Renderer.create(
                <RegistrationForm state={FIXTURES.fetchingState3} actions={FIXTURES.actions} />
            );
            expect(component.toJSON()).toMatchSnapshot();
        });

        it('### render with stable states', () => {
            const component = Renderer.create(
                <RegistrationForm state={FIXTURES.fetchedState} actions={FIXTURES.actions} />
            );
            expect(component.toJSON()).toMatchSnapshot();
        });

        it('### render with form methods', () => {
            let component = Renderer.create(
                <RegistrationForm state={FIXTURES.fetchedState} actions={FIXTURES.actions} />
            );

            let instance = component.getInstance();

            instance.handleEmail(FIXTURES.emailEvent);
            expect(component.toJSON()).toMatchSnapshot();

            instance.dataCallback('test captcha hash');
            expect(component.toJSON()).toMatchSnapshot();

            component = Renderer.create(<RegistrationForm state={FIXTURES.fetchedState} actions={FIXTURES.actions} />);
            instance = component.getInstance();

            instance.dataCallback('test captcha hash');
            expect(component.toJSON()).toMatchSnapshot();

            instance.handleEmail(FIXTURES.emailEvent);
            expect(component.toJSON()).toMatchSnapshot();

            component = Renderer.create(<RegistrationForm state={FIXTURES.fetchedState} actions={FIXTURES.actions} />);
            instance = component.getInstance();

            instance.dataExpiredCallback();
            expect(component.toJSON()).toMatchSnapshot();

            instance.handleEmail(FIXTURES.emailEvent);
            expect(component.toJSON()).toMatchSnapshot();
        });

        it('### render submit methods', () => {
            const component = Renderer.create(
                <RegistrationForm state={FIXTURES.fetchedState} actions={FIXTURES.actions} />
            );
            const instance = component.getInstance();
            instance.onSubmit(FIXTURES.submitEvent);
            expect(component.toJSON()).toMatchSnapshot();
            expect(FIXTURES.actions.createRegistration).toBeCalled();
            expect(FIXTURES.submitEvent.preventDefault).toBeCalled();
        });
    });
});
