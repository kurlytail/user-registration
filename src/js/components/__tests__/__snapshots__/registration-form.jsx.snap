// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`# registration-form ## constructor ### should create default object 1`] = `
<form
  className="auth-form"
  encType="utf8"
  id="user-registration-form"
  method="POST"
>
  <div
    className="form-group"
  >
    <div
      className="form-label-group"
    >
      <label>
        email
      </label>
      <input
        autoFocus={true}
        className="form-control"
        id="user-registration-email"
        onChange={[Function]}
        required={true}
        type="email"
      />
    </div>
  </div>
  <div
    className="form-group"
  >
    <div />
  </div>
  <div
    className="form-group"
  >
    <button
      className="btn btn-lg btn-primary btn-block"
      disabled={true}
      id="signup-button"
      onClick={[Function]}
      type="submit"
    >
      Sign Up
    </button>
  </div>
</form>
`;

exports[`# registration-form ## render ### render submit methods 1`] = `
<form
  className="auth-form"
  encType="utf8"
  id="user-registration-form"
  method="POST"
>
  <div
    className="form-group"
  >
    <div
      className="form-label-group"
    >
      <label>
        email
      </label>
      <input
        autoFocus={true}
        className="form-control"
        id="user-registration-email"
        onChange={[Function]}
        required={true}
        type="email"
      />
    </div>
  </div>
  <div
    className="form-group"
  />
  <div
    className="form-group"
  >
    <button
      className="btn btn-lg btn-primary btn-block"
      disabled={true}
      id="signup-button"
      onClick={[Function]}
      type="submit"
    >
      Sign Up
    </button>
  </div>
</form>
`;

exports[`# registration-form ## render ### render with different states 1`] = `
<form
  className="auth-form"
  encType="utf8"
  id="user-registration-form"
  method="POST"
>
  <div
    className="form-group"
  >
    <div
      className="form-label-group"
    >
      <label>
        email
      </label>
      <input
        autoFocus={true}
        className="form-control"
        id="user-registration-email"
        onChange={[Function]}
        required={true}
        type="email"
      />
    </div>
  </div>
  <div
    className="form-group"
  >
    <div />
  </div>
  <div
    className="form-group"
  >
    <button
      className="btn btn-lg btn-primary btn-block"
      disabled={true}
      id="signup-button"
      onClick={[Function]}
      type="submit"
    >
      Sign Up
    </button>
  </div>
</form>
`;

exports[`# registration-form ## render ### render with different states 2`] = `
<form
  className="auth-form"
  encType="utf8"
  id="user-registration-form"
  method="POST"
>
  <div
    className="form-group"
  >
    <div
      className="form-label-group"
    >
      <label>
        email
      </label>
      <input
        autoFocus={true}
        className="form-control"
        id="user-registration-email"
        onChange={[Function]}
        required={true}
        type="email"
      />
    </div>
  </div>
  <div
    className="form-group"
  >
    <div />
  </div>
  <div
    className="form-group"
  >
    <button
      className="btn btn-lg btn-primary btn-block"
      disabled={true}
      id="signup-button"
      onClick={[Function]}
      type="submit"
    >
      Sign Up
    </button>
  </div>
</form>
`;

exports[`# registration-form ## render ### render with different states 3`] = `
<form
  className="auth-form"
  encType="utf8"
  id="user-registration-form"
  method="POST"
>
  <div
    className="form-group"
  >
    <div
      className="form-label-group"
    >
      <label>
        email
      </label>
      <input
        autoFocus={true}
        className="form-control"
        id="user-registration-email"
        onChange={[Function]}
        required={true}
        type="email"
      />
    </div>
  </div>
  <div
    className="form-group"
  >
    <div />
  </div>
  <div
    className="form-group"
  >
    <button
      className="btn btn-lg btn-primary btn-block"
      disabled={true}
      id="signup-button"
      onClick={[Function]}
      type="submit"
    >
      Sign Up
    </button>
  </div>
</form>
`;

exports[`# registration-form ## render ### render with different states 4`] = `
<form
  className="auth-form"
  encType="utf8"
  id="user-registration-form"
  method="POST"
>
  <div
    className="form-group"
  >
    <div
      className="form-label-group"
    >
      <label>
        email
      </label>
      <input
        autoFocus={true}
        className="form-control"
        id="user-registration-email"
        onChange={[Function]}
        required={true}
        type="email"
      />
    </div>
  </div>
  <div
    className="form-group"
  >
    <div />
  </div>
  <div
    className="form-group"
  >
    <button
      className="btn btn-lg btn-primary btn-block"
      disabled={true}
      id="signup-button"
      onClick={[Function]}
      type="submit"
    >
      Sign Up
    </button>
  </div>
</form>
`;

exports[`# registration-form ## render ### render with different states 5`] = `
<form
  className="auth-form"
  encType="utf8"
  id="user-registration-form"
  method="POST"
>
  <div
    className="form-group"
  >
    <div
      className="form-label-group"
    >
      <label>
        email
      </label>
      <input
        autoFocus={true}
        className="form-control"
        id="user-registration-email"
        onChange={[Function]}
        required={true}
        type="email"
      />
    </div>
  </div>
  <div
    className="form-group"
  >
    <div />
  </div>
  <div
    className="form-group"
  >
    <button
      className="btn btn-lg btn-primary btn-block"
      disabled={true}
      id="signup-button"
      onClick={[Function]}
      type="submit"
    >
      Sign Up
    </button>
  </div>
</form>
`;

exports[`# registration-form ## render ### render with form methods 1`] = `
<form
  className="auth-form"
  encType="utf8"
  id="user-registration-form"
  method="POST"
>
  <div
    className="form-group"
  >
    <div
      className="form-label-group"
    >
      <label>
        email
      </label>
      <input
        autoFocus={true}
        className="form-control"
        id="user-registration-email"
        onChange={[Function]}
        required={true}
        type="email"
      />
    </div>
  </div>
  <div
    className="form-group"
  />
  <div
    className="form-group"
  >
    <button
      className="btn btn-lg btn-primary btn-block"
      disabled={true}
      id="signup-button"
      onClick={[Function]}
      type="submit"
    >
      Sign Up
    </button>
  </div>
</form>
`;

exports[`# registration-form ## render ### render with form methods 2`] = `
<form
  className="auth-form"
  encType="utf8"
  id="user-registration-form"
  method="POST"
>
  <div
    className="form-group"
  >
    <div
      className="form-label-group"
    >
      <label>
        email
      </label>
      <input
        autoFocus={true}
        className="form-control"
        id="user-registration-email"
        onChange={[Function]}
        required={true}
        type="email"
      />
    </div>
  </div>
  <div
    className="form-group"
  />
  <div
    className="form-group"
  >
    <button
      className="btn btn-lg btn-primary btn-block"
      disabled={false}
      id="signup-button"
      onClick={[Function]}
      type="submit"
    >
      Sign Up
    </button>
  </div>
</form>
`;

exports[`# registration-form ## render ### render with form methods 3`] = `
<form
  className="auth-form"
  encType="utf8"
  id="user-registration-form"
  method="POST"
>
  <div
    className="form-group"
  >
    <div
      className="form-label-group"
    >
      <label>
        email
      </label>
      <input
        autoFocus={true}
        className="form-control"
        id="user-registration-email"
        onChange={[Function]}
        required={true}
        type="email"
      />
    </div>
  </div>
  <div
    className="form-group"
  />
  <div
    className="form-group"
  >
    <button
      className="btn btn-lg btn-primary btn-block"
      disabled={true}
      id="signup-button"
      onClick={[Function]}
      type="submit"
    >
      Sign Up
    </button>
  </div>
</form>
`;

exports[`# registration-form ## render ### render with form methods 4`] = `
<form
  className="auth-form"
  encType="utf8"
  id="user-registration-form"
  method="POST"
>
  <div
    className="form-group"
  >
    <div
      className="form-label-group"
    >
      <label>
        email
      </label>
      <input
        autoFocus={true}
        className="form-control"
        id="user-registration-email"
        onChange={[Function]}
        required={true}
        type="email"
      />
    </div>
  </div>
  <div
    className="form-group"
  />
  <div
    className="form-group"
  >
    <button
      className="btn btn-lg btn-primary btn-block"
      disabled={false}
      id="signup-button"
      onClick={[Function]}
      type="submit"
    >
      Sign Up
    </button>
  </div>
</form>
`;

exports[`# registration-form ## render ### render with form methods 5`] = `
<form
  className="auth-form"
  encType="utf8"
  id="user-registration-form"
  method="POST"
>
  <div
    className="form-group"
  >
    <div
      className="form-label-group"
    >
      <label>
        email
      </label>
      <input
        autoFocus={true}
        className="form-control"
        id="user-registration-email"
        onChange={[Function]}
        required={true}
        type="email"
      />
    </div>
  </div>
  <div
    className="form-group"
  />
  <div
    className="form-group"
  >
    <button
      className="btn btn-lg btn-primary btn-block"
      disabled={true}
      id="signup-button"
      onClick={[Function]}
      type="submit"
    >
      Sign Up
    </button>
  </div>
</form>
`;

exports[`# registration-form ## render ### render with form methods 6`] = `
<form
  className="auth-form"
  encType="utf8"
  id="user-registration-form"
  method="POST"
>
  <div
    className="form-group"
  >
    <div
      className="form-label-group"
    >
      <label>
        email
      </label>
      <input
        autoFocus={true}
        className="form-control"
        id="user-registration-email"
        onChange={[Function]}
        required={true}
        type="email"
      />
    </div>
  </div>
  <div
    className="form-group"
  />
  <div
    className="form-group"
  >
    <button
      className="btn btn-lg btn-primary btn-block"
      disabled={true}
      id="signup-button"
      onClick={[Function]}
      type="submit"
    >
      Sign Up
    </button>
  </div>
</form>
`;

exports[`# registration-form ## render ### render with stable states 1`] = `
<form
  className="auth-form"
  encType="utf8"
  id="user-registration-form"
  method="POST"
>
  <div
    className="form-group"
  >
    <div
      className="form-label-group"
    >
      <label>
        email
      </label>
      <input
        autoFocus={true}
        className="form-control"
        id="user-registration-email"
        onChange={[Function]}
        required={true}
        type="email"
      />
    </div>
  </div>
  <div
    className="form-group"
  />
  <div
    className="form-group"
  >
    <button
      className="btn btn-lg btn-primary btn-block"
      disabled={true}
      id="signup-button"
      onClick={[Function]}
      type="submit"
    >
      Sign Up
    </button>
  </div>
</form>
`;
